torch/_inductor/fx_utils.py

```diff
@@ -114,44 +130,37 @@ def is_fake_tensor_same(new, old):
                 return True
             return False
 
+        to_process = set()
         for node in self.graph.nodes:
-            if self.hash_node(node) in self.processed_hashes:
+            if (
+                self.hash_node(node) in self.processed_hashes
+                and id(node) not in to_process
+            ):
                 continue
 
-            def is_aten_node(node):
-                return node.op == "call_function" and isinstance(
-                    node.target, torch._ops.OpOverload
-                )
-
-            if not is_aten_node(node):
+            if node.op != "call_function":
                 continue
 
-            processing = [node]
-            while len(processing) > 0:
-                updating_node = processing.pop()
-                if updating_node in processed:
-                    continue
-                if is_aten_node(updating_node):
-                    continue
-
-                is_valid, args, kwargs = get_fake_args_kwargs(updating_node)
-                if not is_valid:
-                    continue
+            is_valid, args, kwargs = get_fake_args_kwargs(node)
+            if not is_valid:
+                continue
+            try:
                 with V.fake_mode:
-                    new_fake_tensor = updating_node.target(*args, **kwargs)
-                if "val" in updating_node.meta and is_fake_tensor_same(
-                    new_fake_tensor, updating_node.meta["val"]
-                ):
-                    continue
-                updating_node.meta["val"] = new_fake_tensor
-
-                # todo(chilli): This code path is not exercised by our existing
-                # tests - add a test
-                existing_storages[get_node_storage(new_fake_tensor)] += 1
-                processed.add(updating_node)
-                processing.extend(updating_node.users)
-
-                self.processed_hashes.add(self.hash_node(updating_node))
+                    new_fake_tensor = node.target(*args, **kwargs)
+            except (ValueError, AttributeError, AssertionError, RuntimeError):
```

Comment:
Okay, so with the added context that lowering patterns don't have fake implementations then I think we should go back to only propagating through `OpOverload` but have and extra carve out for `operator.getitem` to fix the multi-output problem.

Reply:
And it would be good to add a comment that graph lowering patterns are the issue.